test/suite/test_bug029.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_bug029.py
+#       Variable column store search near key past the end.
+
+import wttest
+from wtscenario import make_scenarios
+
+class test_bug029(wttest.WiredTigerTestCase):
+    uri = 'file:test_bug029'
+
+    key_format_values = [
+        ('var', dict(key_format='r', value_format='I')),
+        ('row', dict(key_format='Q', value_format='I')),
+    ]
```

Comment:
We can also add fixed length column store.

Reply:
Won't the delete scenario fail then?

Reply:
We need to write different validation code for that.

Reply:
The way I've been doing it is just self.value_format == '8t'. Not ideal, but serviceable, and enough of them are that way that even if someone decides they hate it another won't be noticed :-)

Reply:
Oh, also, bug029 is a fine name but ISTM you might reasonably call it search_near05. Will make it easier to find later...

Reply:
Thanks, I've fixed this (and the naming) in 84fdeae.

Reply:
Looks good.